libraries/botbuilder-dialogs-adaptive/src/actions/beginSkill.ts

```diff
@@ -182,7 +182,7 @@ export class BeginSkill extends SkillDialog implements BeginSkillConfiguration {
             this.dialogOptions.connectionName = this.connectionName.getValue(dcState);
         }
         if (!this.dialogOptions.conversationState) {
-            this.dialogOptions.conversationState = dc.dialogManager.conversationState;
+            this.dialogOptions.conversationState = dc.context.turnState.get<ConversationState>('ConversationState');
```

Comment:
Intent is to reconcile this with https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/BeginSkill.cs#L159

Reply:
Should the JS impl throw an Exception like in .NET if a ConversationState instance isn't retrieved from the TurnState?

Reply:
Good question - I was following the existing implementation with respect to throwing. Should it?

Reply:
From what I can tell it definitely should; without the SkillClient, ConversationIdFactory and ConversationState the host bot won't be able to send messages, process messages or update the conversationState before forwarding activities 😅

Reply:
If beginSkill somehow works without the exceptions, then we have a diff that needs to be unpacked @chon219 have you used `BeginSkill` in Composer/AdaptiveDialogs testing?

Reply:
We do have tests using `BeginSkill` but I have not tested `BeginSkill` without conversationState, in fact, `DialogManager` will throw exception in its `onTurn` if there is no conversationState found in turn state.

Reply:
Seems like we protect against this case further up the chain so I don't think we need another error thrown.

Reply:
Overall I'm okay with this, though BeginSkill is a public class and can be used outside of DialogManager. The belief that the general code-first population will not use AdaptiveDialogs and by extension BeginSkill is why I'm not opposed to this.

We should be evaluate porting these changes back to .NET or keeping a record of the difference in behavior between the implementations.